thrust/testing/transform_iterator.cu

```diff
@@ -108,3 +108,33 @@ void TestTransformIteratorNonCopyable()
 }
 
 DECLARE_UNITTEST(TestTransformIteratorNonCopyable);
+
+struct foo
+{
+  int x, y;
+};
+
+struct access_x
+{
+  _CCCL_HOST_DEVICE int& operator()(foo& f) const noexcept
+  {
+    return f.x;
+  }
+};
+
+void TestTransformIteratorAsDestination()
+{
+  constexpr auto n = 10;
+  thrust::host_vector<int> src(n, 1234);
```

Comment:
Suggestion: Also test with `device_vector`. The CUDA-backed `device_vector` references/pointers/etc are significantly more complex than the `host_vector` implementations.